### PR TITLE
feat: Add views and access for internal reviewer role

### PR DESCRIPTION
**Pull request by martin-trajanovski**
## Description

This PR aims to extend the new feature called `Internal Reviews` with views for the internal reviewers. Those are reviews that are used only internally as a help for the technical review on a proposal. We can have as many internal reviews as we want per technical review on a proposal. There is even a new role introduced called `INTERNAL_REVIEWER` which is used to filter people assigned to the internal review.

## Motivation and Context

Previously (in the old way and older software) this was more or less done by phone as a process of consulting more people/reviewers about their opinion about the technical review of a proposal. This is much better way because we keep track of it in the system. The reviews are not used on a proposal later as part of it's pdf or anything similar.

## How Has This Been Tested

- manual tests

## Fixes

https://jira.esss.lu.se/browse/SWAP-3383

https://github.com/UserOfficeProject/user-office-core/assets/6333063/2b060085-2868-4d22-8a23-ebf455580787

## Tests included/Docs Updated?

- e2e tests are coming in a separate PR: https://github.com/UserOfficeProject/user-office-core/pull/284

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

**Comment by enthusiastio** [comment]
I noticed one thing, is that if as a user officer, you log in, and for the proposal that has no review, add a review, then immediately after the internal review component is shown. If clicked it goes into an infinite loop, until the app is refreshed. Then it can fetch it
Peek 2023-08-02 20-09
Maybe it is something you want to address

**Comment by enthusiastio** [comment]
I also wonder what is the purpose of the title. Do we need it?
image

Maybe there should rather be a comment for the reviewer?

**Comment by enthusiastio** [comment]
On this component that the reviewer is using to add his internal review, maybe we should not show disabled input fields. The user probably also knows his name :), so could get rid of that, and then the title, if needed could be just a header. I think you are reusing the component from user officer, but maybe it should just be a new component for reviewer

image

File: apps/user-office-backend/src/datasources/InternalReviewDataSource.ts [hunk 1]
```diff
@@ -17,4 +17,10 @@ export interface InternalReviewDataSource {
     input: UpdateInternalReviewInput
   ): Promise<InternalReview>;
   delete(input: DeleteInternalReviewInput): Promise<InternalReview>;
+  isInternalReviewerOnTechnicalReview(
+    userId: number,
+    technicalReviewId: number
+  ): Promise<boolean>;
+  isInternalReviewer(userId: number): Promise<boolean>;
+  getRelatedUsersOnInternalReview(id: number): Promise<number[]>;
```
[the remark this answers is not in the document]
Comment (martin-trajanovski) [reply apps/user-office-backend/src/datasources/InternalReviewDataSource.ts:25]: @enthusiastio thanks for the extensive testing. I will try to resolve all those issues and ask you to review again.

**Comment by martin-trajanovski** [comment]
> I also wonder what is the purpose of the title. Do we need it? image
> 
> Maybe there should rather be a comment for the reviewer?

Yes we discussed with Fredrik and Carina about that. The title is used to name the review so the one who requested it knows later what is it about and why is requested. Something like reminder and for the reviewers to know what is requested from them.

**Comment by martin-trajanovski** [comment]
@enthusiastio now I tried to resolve all the issues that were mentioned above and you can double check this to verify. Thanks once more for the good review and comments.

**Review by enthusiastio — APPROVED** [review]
:+1:

**Review by TCMeldrum — APPROVED** [review]
Looks good to me. I think we will want to use this feature at some point but we will need to work out how we assign the `INTERNAL_REVIEWER` and we will need to rework our implantation of `getInstrumentScientistProposals` but we can work on those separate from this PR.

**Review by mutambaraf — APPROVED** [review]

**Comment by martin-trajanovski** [comment]
> Looks good to me. I think we will want to use this feature at some point but we will need to work out how we assign the `INTERNAL_REVIEWER` and we will need to rework our implantation of `getInstrumentScientistProposals` but we can work on those separate from this PR.

@TCMeldrum thanks for the review and I see that you have created a separate issue opening this question on your side which is great.

**Pull request merged**